packages/catalog-process/src/services/catalogService.ts

```diff
@@ -537,6 +539,64 @@ export function catalogServiceBuilder(
         return retrieveDocument(eServiceId, descriptor, documentId);
       }
     },
+
+    async getEServices(
+      authData: AuthData,
+      {
+        eservicesIds,
+        producersIds,
+        states,
+        agreementStates,
+        name,
+      }: {
+        eservicesIds: string[];
+        producersIds: string[];
+        states: DescriptorState[];
+        agreementStates: AgreementState[];
+        name?: string;
+      },
+      offset: number,
+      limit: number
+    ): Promise<ListResult<EService>> {
+      const eservicesList = await readModelService.getEServices(
+        authData,
+        { eservicesIds, producersIds, states, agreementStates, name },
+        offset,
+        limit
+      );
+      let eServicesToReturn: EService[]; // eslint-disable-line functional/no-let
+      if (
+        !(
+          authData.userRoles.includes(userRoles.ADMIN_ROLE) ||
+          authData.userRoles.includes(userRoles.API_ROLE)
+        )
+      ) {
+        eServicesToReturn = eservicesList.results.map((eService) => ({
+          ...eService,
+          descriptors: eService.descriptors.filter(
+            (d) => d.state !== descriptorState.draft
+          ),
+        }));
+      } else {
+        eServicesToReturn = eservicesList.results.map((eService) => {
+          if (eService.producerId !== authData.organizationId) {
+            return {
+              ...eService,
+              descriptors: eService.descriptors.filter(
+                (d) => d.state !== descriptorState.draft
+              ),
+            };
+          } else {
+            return eService;
+          }
+        });
+      }
```

Comment:
This logic could be improved by removing the let and moving the if statements inside the map function.
My proposal:

```suggestion
      const eServicesToReturn: EService[] = eservicesList.results.map(
        (eService) => {
          const shouldIncludeDraftDescriptors =
            eService.producerId !== authData.organizationId &&
            (authData.userRoles.includes(userRoles.ADMIN_ROLE) ||
              authData.userRoles.includes(userRoles.API_ROLE));

          if (!shouldIncludeDraftDescriptors) {
            return {
              ...eService,
              descriptors: eService.descriptors.filter(
                (d) => d.state !== descriptorState.draft
              ),
            };
          }
          return eService;
        }
      );
```

Reply:
Thanks for the suggestion! We could even avoid to wrap the conditions in a custom variable.